data/binding/printf.go

```diff
@@ -0,0 +1,254 @@
+package binding
+
+import (
+	"fmt"
+)
+
+type sprintfString struct {
+	String
+
+	format string
+	source []interface{}
+	err    error
+}
+
+// NewSprintf returns a String binding that format its content using the
+// format string and the provide additional parameter that must be other
+// data bindings.
+//
+// Since: 2.2
+func NewSprintf(format string, b ...interface{}) (String, error) {
+	ret := &sprintfString{String: NewString(), format: format, source: make([]interface{}, 0)}
```

Comment:
Slices don't need to be initialized before calling `append`
```suggestion
	ret := &sprintfString{String: NewString(), format: format}
```